airbyte-cdk/python/airbyte_cdk/sources/streams/http/http_client.py

```diff
@@ -259,9 +259,15 @@ def _send(
         # Evaluation of response.text can be heavy, for example, if streaming a large response
         # Do it only in debug mode
         if self._logger.isEnabledFor(logging.DEBUG) and response is not None:
-            self._logger.debug(
-                "Receiving response", extra={"headers": response.headers, "status": response.status_code, "body": response.text}
-            )
+            if request_kwargs.get("stream"):
```

Comment:
what does this check verify in practice?

Reply:
this is done in order to remove response.text from debug message if response is streamable